### PR TITLE
Add progress bar for parsing files, use Path to work with files

### PR DESCRIPTION
**Pull request by myakove**
1. Add a progress bar when parsing must-gather files
2. Use `Path` instead `os` to work with files.
3. Add --verbose to display debug logs
4. small code refactors

## Summary by CodeRabbit

- **New Features**
	- Introduced a `verbose` option for detailed logging in the command-line interface.
	- Enhanced error handling for reading aliases files, providing clearer feedback on errors.

- **Bug Fixes**
	- Improved error handling for YAML file parsing to prevent program termination.

- **Documentation**
	- Updated README to include instructions for using the `verbose` flag for debugging logs.

- **Chores**
	- Updated dependencies in the project configuration to include new logging and progress tracking libraries.

### CODE REVIEW

**Comment by redhat-qe-bot1** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install
 * `/retest all`: Retest all

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

File: app/main.py [hunk 1]
```diff
@@ -11,70 +13,101 @@
...
+
+                file_extention = _file.suffix
+                if not file_extention:
+                    LOGGER.debug(f"Skipping file {_file}, file has no extension")
+                    continue
+
+                if file_extention in (".yaml", ".yml"):
+                    all_yaml_files.setdefault(str(_dir_path), []).append(_file.name)
 
                 # Poa may have several containers, the path to the log is:
```
Comment (jpeimer) [inline app/main.py:65]: Could you please fix to `# Pod may have several containers`

File: app/main.py [hunk 2]
```diff
@@ -11,70 +13,101 @@
...
+        dynamic_ncols=True,
+    ):
         for yaml_file in yaml_files:
-            yaml_file_path = os.path.join(yaml_path, yaml_file)
+            yaml_file_path = Path(yaml_path, yaml_file)
+
             with open(yaml_file_path) as fd:
-                resource_dictionary = yaml.safe_load(fd)
+                try:
+                    resource_dictionary = yaml.safe_load(fd)
```
Comment (jpeimer) [inline app/main.py:86]: Should `with open` go under the `try` as well?

**Review by jpeimer — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -11,70 +13,101 @@
...
+    required=True,
     help="""
     \b
     The full path to the must-gather folder
 """,
 )
-def main(
-    path: str,
-) -> None:
+@click.option("-v", "--verbose", is_flag=True, help="Enable verbose logs")
```
[the remark this answers is not in the document]
Comment (myakove) [reply app/main.py:32]: Done

File: app/main.py (hunk 2 above)
Comment (myakove) [reply app/main.py:86]: No, we collect those yamls, so we know it's there

File: app/main.py (hunk 1 above)
Comment (myakove) [reply app/main.py:65]: Done

File: app/main.py [hunk 4]
```diff
@@ -11,70 +13,101 @@
...
+                    continue
+
             resource_dict_metadata = resource_dictionary["metadata"]
-            all_resources.setdefault(f"{resource_dictionary['kind']}".lower(), []).append({
+            all_resources.setdefault(resource_dictionary["kind"].lower(), []).append({
                 "name": resource_dict_metadata.get("name", ""),
                 "namespace": resource_dict_metadata.get("namespace", ""),
                 "yaml_file": yaml_file_path,
             })
+
```
[the remark this answers is not in the document]
Comment (jpeimer) [reply app/main.py:97]: We expect all yamls from must-gather to have 'kind' and 'metadata' keys. If they are not present, then probably something is wrong with the must-gather path or content. We may want to handle it as a separate issue.

**Pull request merged**